website/blog/2023-12-23-AgentOptimizer/index.mdx

```diff
@@ -0,0 +1,179 @@
+                },
+                "description": {
+                    "type": "string",
+                    "description": "A short description of the function."
+                },
+                "arguments": {
+                    "type": "string",
+                    "description": "JSON schema of arguments encoded as a string. Please note that the JSON schema only supports specific types including string, integer, object, array, boolean. (do not have float type) For example: { \"url\": { \"type\": \"string\", \"description\": \"The URL\", }}. Please avoid the error 'array schema missing items' when using array type."
+                },
+                "packages": {
+                    "type": "string",
+                    "description": "A list of package names imported by the function, and that need to be installed with pip prior to invoking the function. This solves ModuleNotFoundError. It should be string, not list."
+                },
+                "code": {
+                    "type": "string",
+                    "description": "The implementation in Python. Do not include the function declaration."
+                }
+            },
+            "required": ["name", "description", "arguments", "packages", "code"]
+        }
+    }
+}
+```
+
+3. `remove_function`: Remove one existing function in the current function list. It is used to remove the functions that are not useful (redundant) in the future tasks.
+```python
+REMOVE_FUNC = {
+    "type": "function",
+    "function": {
+        "name": "remove_function",
+        "description": "Remove one function in the context of the conversation. Once remove one function, the assistant will not use this function in future conversation.",
+        "parameters": {
+            "type": "object",
+            "properties": {
+                "name": {
+                    "type": "string",
+                    "description": "The name of the function in the code implementation."
+                }
+            },
+            "required": ["name"]
+        }
+    }
+}
+```
+
+
+## Limitation & Future work
+
+1. Unlike gradient descent in traditional machine learning training processes, each optimization step does not necessarily lead to better performance on the training set.
+When the training epoch is small, the agent’s performance may even decrease. One urgent task is to design a better mechanism to guide the optimization process.
+2. The [Current implementation](https://github.com/microsoft/autogen/blob/main/notebook/agent_optimizer.ipynb) of AgentOptimizer is mainly for illustration purpose and is just a proof of concept.
+It is not formally integrated into the autogen with a general interface like optimizing any kinds of agents in any tasks.
+Currently, it only supports the optimization the agents in [MATH](https://github.com/hendrycks/math) dataset. We will integrate it into autogen with more general interface in the future.
```

Comment:
This description "the agents in MATH dataset" is not a proper description of the applicable agents (actually a multi-agent system)

Reply:
revised.